cognite/client/utils/_time.py

```diff
@@ -550,24 +552,39 @@ def pandas_date_range_tz(start: datetime, end: datetime, freq: str, inclusive: s
     )
 
 
+def _timezones_are_equal(start_tz: tzinfo, end_tz: tzinfo) -> bool:
+    """There are unfortunately several ways to pass/represent the same timezone (without it being a user error).
+    For example pandas uses 'pytz' under the hood -except- for UTC, then it uses the built-in `datetime.timezone.utc`
+    -except- when given something concrete like pytz.UTC or ZoneInfo(...).
+
+    To make sure we don't raise something silly like 'UTC != UTC', we convert both to ZoneInfo for comparison
+    via str(). This is safe as all return the lookup key (for the IANA time zone database).
+
+    Note:
+        We do not consider timezones with different keys, but equal fixed offsets from UTC to be equal. An example
+        would be Zulu Time (which is +00:00 ahead of UTC) and UTC.
+    """
+    if start_tz is end_tz:
+        return True
+    ZoneInfo, ZoneInfoNotFoundError = import_zoneinfo(), _import_zoneinfo_not_found_error()
+    with suppress(ValueError, ZoneInfoNotFoundError):
+        # ValueError is raised for non-conforming keys (ZoneInfoNotFoundError is self-explanatory)
+        if ZoneInfo(str(start_tz)) is ZoneInfo(str(end_tz)):
+            return True
+    return False
```

Comment:
🏌
```suggestion
        return ZoneInfo(str(start_tz)) is ZoneInfo(str(end_tz))
    return False
```

Reply:
That doesn’t work as it will return `None` instead of `False` when one of the suppressed errors occur.